py/kubeflow/testing/run_e2e_workflow.py

```diff
@@ -348,21 +359,39 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
                get_namespace(args)], cwd=w.app_dir)
       util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "bucket",
                args.bucket], cwd=w.app_dir)
+      util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "cluster_name",
+                "eks-cluster-{}".format(salt)], cwd=w.app_dir)
+      util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "desired_node",
+               args.desired_node], cwd=w.app_dir)
+      util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "min_node",
+               args.min_node], cwd=w.app_dir)
+      util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "max_node",
+               args.max_node], cwd=w.app_dir)
       if args.release:
         util.run([ks_cmd, "param", "set", "--env=" + env, w.component, "versionTag",
                   os.getenv("VERSION_TAG")], cwd=w.app_dir)
 
       # Set any extra params. We do this in alphabetical order to make it easier to verify in
       # the unittest.
       param_names = w.params.keys()
-      param_names.sort()
+      # In python3, dict_keys.sort() not work given
+      # https://docs.python.org/3/whatsnew/3.0.html#views-and-iterators-instead-of-lists
+      param_names = sorted(param_names)
       for k in param_names:
         util.run([ks_cmd, "param", "set", "--env=" + env, w.component, k,
                  "{0}".format(w.params[k])], cwd=w.app_dir)
 
       # For debugging print out the manifest
       util.run([ks_cmd, "show", env, "-c", w.component], cwd=w.app_dir)
-      util.run([ks_cmd, "apply", env, "-c", w.component], cwd=w.app_dir)
+
+      if not args.cloud_provider:
+        util.run([ks_cmd, "apply", env, "-c", w.component], cwd=w.app_dir)
+      elif args.cloud_provider == "aws":
+        generated_workflow_name = "generated_workflow.yaml"
+        util.save_process_output([ks_cmd, "show", env, "-c", w.component],
```

Comment:
Any reason not to use kf submit pipeline directly?

Reply:
Fixed